### PR TITLE
feat(RangeSlider): Improve keyboard control

### LINKED ISSUE #1429: RangeSlider Keyboard Control
Add keyboard control to RangeSlider to improve accessibility.

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #1429

## Overview

~~Use `button` instead of `div` for handle. Improve keyboard control.~~

Apply `tabIndex` to Handles and maintain `div` element due to accessibility AXE role error.

## Reason

>There were some weird edge cases where the handle wouldn't be controllable via the keyboard. Despite automated tests that were passing.

## Work carried out

- [x] Change handle element

### CODE REVIEW

File: packages/react-component-library/src/components/RangeSlider/Handle.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const Handle: React.FC<HandleProps> = ({
   })
 
   return (
-    <div
+    <button
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RangeSlider/Handle.tsx]: This was changed recently. You will notice the a11y error:

```
Rule: "aria-allowed-role" (ARIA role must be appropriate for the element)
```

Comment (m7kvqbe1) [reply packages/react-component-library/src/components/RangeSlider/Handle.tsx]: I've reverted back to the `div` and resolved the issue by applying `tabIndex` to handles.

**Review by thyhjwb6 — APPROVED** [review]
